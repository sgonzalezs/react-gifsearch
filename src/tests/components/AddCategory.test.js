import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';


describe('Tests file AddCategory.js', () => {

    const setCategories=jest.fn();
    let wrapper=shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper=shallow(<AddCategory setCategories={setCategories} />);
    });

    test('test Snapshot component', () => {
        
        expect(wrapper).toMatchSnapshot();
        
    }); 

    test('must change the input', () => {
        
        const input=wrapper.find('input');
        const value='Technology';
        input.simulate('change', {target:{value}});

        expect(wrapper.find('p').text().trim()).toBe(value);

    });

    test('NO post submit', () => {
       wrapper.find('form').simulate('submit', {preventDefault:()=>{}}); 
       expect(setCategories).not.toHaveBeenCalled();
    });

    test('Must call setCategories and clean the textbox', () => {

        const value='Movies';
        wrapper.find('input').simulate('change', {target:{value}});
        
        wrapper.find('form').simulate('submit', {preventDefault:()=>{}});

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);

        // Validamos que se haya llamado el metodo setCategories con una funcion
        expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );

        expect(wrapper.find('input').prop('value')).toBe('');

    });
    
});
