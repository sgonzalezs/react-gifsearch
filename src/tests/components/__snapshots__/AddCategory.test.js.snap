// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests file AddCategory.js test Snapshot component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
